nomad/rpc.go

```diff
@@ -107,6 +107,38 @@ type RPCContext struct {
 	NodeID string
 }
 
+// Certificate returns the first certificate available in the chain.
+func (ctx *RPCContext) Certificate() *x509.Certificate {
```

Comment:
ValidateCertificateForName guards against `ctx` itself being nil and we may want to do the same here.